### PR TITLE
Useful page load

### PR DESCRIPTION
**Pull request by andrewconner**

### CODE REVIEW

File: shoebox/app/com/keepit/common/analytics/EventListener.scala [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+import play.api.libs.json.{JsArray, JsBoolean, JsNumber, JsObject, JsString}
+import com.keepit.model._
+import com.keepit.common.db._
+import com.keepit.search.ArticleSearchResultRef
+import play.api.Play.current
+import java.sql.Connection
+
+object EventListener {
+  val listeners: Seq[EventListener] = Seq(KifiResultClickedListener, UsefulPageListener, DeadQueryListener)
```
Comment (eishay) [inline shoebox/app/com/keepit/common/analytics/EventListener.scala:11]: I don't like the scala say of doing singletons, i.e. having val or var in an object. In fact, I don't like scala object too much :-)
Please check out http://code.google.com/p/google-guice/wiki/Multibindings
Please use it instead. Pull me in if you need help, I'll be happy. I think the example is good enough.

Comment (andrewconner) [inline shoebox/app/com/keepit/common/analytics/EventListener.scala:11]: Thanks for the feedback. I'll check it out and implement.

File: shoebox/app/com/keepit/common/analytics/EventListener.scala [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      val url = (json \ "url").asOpt[String].getOrElse("")
+      val user = User.get(externalUser)
+      val normUrl = NormalizedURI.getByNormalizedUrl(url)
+      val queryUUID = ExternalId.asOpt[ArticleSearchResultRef]((json \ "queryUUID").asOpt[String].getOrElse(""))
+      (user, SearchMeta(query, url, normUrl, queryUUID))
+  }
+}
+
+
+object KifiResultClickedListener extends EventListener {
```
Comment (eishay) [inline shoebox/app/com/keepit/common/analytics/EventListener.scala:36]: Event listeners should be classes, not objects. You can't mock them this way

File: shoebox/app/com/keepit/common/analytics/EventLog.scala [hunk 3]
```diff
@@ -69,6 +69,9 @@ case class ServerEventMetadata(eventFamily: EventFamily, eventName: String, meta
 
 case class Event(externalId: ExternalId[Event] = ExternalId[Event](), metaData: EventMetadata, createdAt: DateTime,
     serverVersion: String = inject[FortyTwoServices].currentService + ":" + inject[FortyTwoServices].currentVersion) {
+
+  EventListener.newEvent(this)
```
Comment (eishay) [inline shoebox/app/com/keepit/common/analytics/EventLog.scala:73]: Please inject the EventListener, don't use a singleton pattern to get it. Its bad for testing and mocking.
Use

``` scala
inject[EventListener].newEvent(this)
```

File: shoebox/app/com/keepit/common/analytics/reports/ReportBuilder.scala [hunk 4]
```diff
@@ -40,9 +40,10 @@ object Reports {
   lazy val DailySliderClosedByIcon = new DailySliderClosedByIcon
   lazy val DailySliderClosedByKey = new DailySliderClosedByKey
   lazy val DailySliderClosedByX = new DailySliderClosedByX
-  lazy val DailyNewComment = new DailyNewComment
-  lazy val DailyNewMessage = new DailyNewMessage
-  lazy val DailyNewUnkeep = new DailyNewUnkeep
+  lazy val DailyComment = new DailyComment
```
Comment (eishay) [inline shoebox/app/com/keepit/common/analytics/reports/ReportBuilder.scala:43]: Can you please remind me why the vals here starts with capital letter?

Comment (andrewconner) [inline shoebox/app/com/keepit/common/analytics/reports/ReportBuilder.scala:43]: Nope :). It should be camelCaseLikeEverythingElse.

**Pull request merged**